src/soundvectors.py

```diff
@@ -406,125 +506,108 @@ def validate(self, sound):
                     return sound_transcribed.name
         raise ValueError("Invalid sound encountered.")
 
-    def get_vec(self, sound, vectorize=True):
-        """
-        Retrieve the vector for a given sound.
-
-        @param sound: A sound represented by a feature string. If a
-            transcription system has been supplied, it can also be a 
-            sound in IPA notation.
-        @param vectorize: Return a binary vector if set to True.
-        @returns: vector (dictionary of feature values or tuple of values)
-        
-        Examples::
-            
-            >>> from soundvectors import SoundVectors
-            >>> sv = SoundVectors()
-            >>> vec = sv.get_vec("t", vectorize=False)
-            >>> print(vec["cont"])
-            - 1
-        """
+    def __getitem__(self, sound: typing.Union[str, Sound]) -> FeatureBundle:
         sound = self.validate(sound)
-        base_vec = FeatureBundle(self.binary_features)  # {f: 0 for f in self.binary_features}
-
-        # check if sound is diphthong or complex
-        complex_sound = False
-        if sound.endswith(" diphthong"):
-            self._apply_feature("diphthong", base_vec)
-            sound_to_sound = sound[sound.index(" to ") + 4: -10] + " vowel"
-            sound = sound[5: sound.index(" to ")] + " vowel"
-            # set up feature pairs for diphthong trajectory
-            diphthong_features = (["from_" + f for f in sound.split()[:-1]] +
-                                  ["to_" + f for f in sound_to_sound.split()[:-1]])
-            complex_sound = "diphthong"
-
-        elif sound.endswith(" cluster"):
-            sound_to_sound = sound[sound.index(" to ") + 4: -8] + " consonant"
-            sound = sound[5: sound.index(" to ")] + " consonant"
-            complex_sound = "cluster"
+        base_vec = FeatureBundle()
+
+        # check if sound is complex
+        complex_sound = sound.split()[-1] if sound.split()[-1] in COMPLEX_SOUNDS else False
+        if complex_sound:
+            from_sound, complex, to_sound = sound.partition(' to ')
+            assert complex
+            sound_to_sound = to_sound.replace(complex_sound, COMPLEX_SOUNDS[complex_sound])
+            sound = '{} {}'.format(from_sound, COMPLEX_SOUNDS[complex_sound])
+
+            if complex_sound == "diphthong":
+                base_vec = self._apply_feature("diphthong", base_vec)
+                # set up feature pairs for diphthong trajectory
+                diphthong_features = (["from_" + f for f in sound.split()[:-1]] +  # noqa: W504
+                                      ["to_" + f for f in sound_to_sound.split()[:-1]])
 
         features = [f for df in sound.split() for f in df.split("-and-")]
 
         primary_features, secondary_features = self._get_features(features)
 
         for feature in primary_features:
-            self._apply_feature(feature, base_vec)
+            base_vec = self._apply_feature(feature, base_vec)
```

Comment:
Since `FeatureBundle` instances are immutable now, the `_apply*` methods must return the new instance rather than manipulate the input in place.